datajs/src/data.js

```diff
@@ -0,0 +1,239 @@
+import fs from 'fs'
+import path from 'path'
+import url from 'url'
+import fetch from 'node-fetch'
+import { cloneDeep, isPlainObject, isString } from 'lodash'
+import mime from 'mime-types'
+import { csvParser } from './parser/csv'
+import { xlsxParser } from './parser/xlsx'
+
+// for browser related functions
+// import browser from './browser-utils/index'
+import { isFileFromBrowser } from './browser-utils/index'
+
+
+import { FileInterface } from './file-base'
+import { FileLocal } from './file-local'
+import { FileRemote } from './file-remote'
+import { FileInline } from './file-inline'
+
+export const DEFAULT_ENCODING = 'utf-8'
+
+// Available parsers per file format
+export const PARSE_DATABASE = {
+    csv: csvParser,
+    tsv: csvParser,
+    xlsx: xlsxParser,
+    xls: xlsxParser,
+}
+
+// List of formats that are known as tabular
+export const KNOWN_TABULAR_FORMAT = ['csv', 'tsv', 'dsv']
+
+
+/**
+ * create a File from a pathOrDescriptor
+ * @param {string} pathOrDescriptor 
+ * @param {object} options 
+ */
+export function open(pathOrDescriptor, { basePath, format } = {}) {
+    let descriptor = null
```

Comment:
we should have 2-space indentation in JS. Can we set linter for this and fix that in these files. See https://playbook.datopian.com/style-guide/javascript/

Reply:
Okay, will do that now

Reply:
Fixed, pushed new changes already